doc/reference/trd-appid.md

```diff
@@ -379,50 +379,76 @@ deterministic behavior in the presence of colliding Application
-does not have a TBF Program Header.
+Using the provided Credentials Checking Policy (described in Section
+6), it decides whether each process has permission to run. It moves
+processes whose TBF Objects are not allowed to run into the
+`CredentialsFailed` state. It moves processes whose TBF Objects are
+allowed to run into the `CredentialsApproved` state. Moving a process
+into the `CredentialsApproved` state triggers the main kernel loop to
+examine the process.
+
+The `CredentialsApproved` state indicates that the process is runnable
+in terms of its credentials. However, at any given time it might not
+be allowed to run because its Application Identifier or Short ID
+conflicts with another processs. When the main kernel loop examines a
+process in the `CredentialsApproved` state, it determines whether to
+run the process based on its Application Identifier, Short ID, and the
+Application Binary version number (stored in the Program Header,
+described in Section 5.1). At boot, the kernel starts a process if:
+
+  - The process has a unique Application Identifier and Short ID,
+  - The process has a higher Application Binary version number than
+    all processes it shares its Application Identifier or Short ID with,
+
+If two processes which share a Short ID or Application ID have the
+same version number, the kernel starts one of them. The one which
+starts is determined by the scheduling policy of the kernel (whichever
+one is run first).
 
 The Unloaded, CredentialsUnchecked, CredentialsFailed,
-CredentialsPassed and Running states describe the conceptual state
+CredentialsApproved and Running states describe the conceptual state
 machine of a process at loading; the values or names of state
 variables in the kernel implemementation might be different. The exact
 Tock process state machine and names of its states is outside the
 scope of this document.
 
+One a Tock system is running, management interfaces may change the set
+of running processes from those which the boot sequence
+selected. E.g., the process console might terminate a process so that
+it can run a different process with the same Short ID and a lower
+Userspace Binary version number (rollback). The kernel maintains that
+a running process has a unique Application Identifier and a unique
+Short ID among running processes.
+
+Note that the algorithm above can lead to transitive blocking, such
+that a process at boot is blocked from running even if the process
+that blocks it does not run. For example, suppose there are three
+processes: A, B and C.
+
+  - A has a version number of 1, an Application ID of X and a Short ID of 6
+  - B has a version number of 2, an Application ID of Y and a Short ID of 6
+  - C has a version number of 3, an APplication ID of Y and a Short ID of 7
+
+Following the above logic, A cannot run because it shares a Short ID
+with A and has a lower version number. At the same time, B cannot run
```

Comment:
```suggestion
Following the above logic, A cannot run because it shares a Short ID
with B and has a lower version number. At the same time, B cannot run
```